### PR TITLE
Add attr  and name to idata

### PR DESCRIPTION
**Pull request by percygautam**
## Description
Related to the discussion being held in the channel.

## Checklist

- [ ] Follows [official](https://github.com/arviz-devs/arviz/blob/master/CONTRIBUTING.md#pull-request-checklist) PR format
- [ ] Includes a sample plot to visually illustrate the changes (only for plot-related functions)
- [x] New features are properly documented (with an example if appropriate)?
- [x] Includes new or updated tests to cover the new feature
- [x] Code style  correct (follows pylint and black guidelines)
- [ ] Changes are listed in [changelog](https://github.com/arviz-devs/arviz/blob/master/CHANGELOG.md#v0xx-unreleased)

### CODE REVIEW

**Comment by percygautam** [comment]
How these changes are incorporated in `concat`. Also, briefly explain the `name` attribute?

**Comment by ahartikainen** [comment]
For concat, either we make everything a list or list only when needed, same for name.

    idata1.attrs["a"] = 1
    idata2.attrs["a"] = 1

    -> idata["a"] == [1, 1] or idata["a"] == 1

    idata1.attrs["a"] = 1
    idata2.attrs["a"] = 2

    -> idata["a"] == [1, 2]

**Comment by ahartikainen** [comment]
I think that might be the simplest option?

**Comment by percygautam** [comment]
I have added `attrs` to concat as suggested. I was modifying `from_dict` to include attrs, but it already contains `attrs` argument, that point to `dataset.attrs`. Should I change already existing attrs to `dataset_attrs`? Or something else?

**Comment by ahartikainen** [comment]
I think attrs in from_dict should point to the main attrs (this what you add right now)

**Comment by percygautam** [comment]
Will this not change the existing API? Also, if we do so, what should the `attrs` be named? (Maybe `group_attrs` or `dataset_attrs`?)

**Comment by ahartikainen** [comment]
I think existing api is not yet released?

That sounds a good idea.

We could also add kwargs for `from_dict` and then get attrs for each group if needed?

So `attrs` could then mean the idata attrs. And each group might have `{groupname}_attrs`?

**Comment by OriolAbril** [comment]
Another question is whether we should modify current code to store arviz/ppl version... as idata attrs instead of repeating them in every group.

File: arviz/data/io_dict.py [hunk 1]
```diff
@@ -335,6 +368,8 @@ def from_dict(
         A mapping from variables to a list of coordinate values for predictions.
     attrs : dict
         A dictionary containing attributes for different groups.
+    kwargs : dict
+        A dictionary containing group attrs.
```
Comment (OriolAbril) [inline arviz/data/io_dict.py:372]: Accepted kwargs should be docummented.

**Comment by ahartikainen** [comment]
Do we want to get this in the next release?

**Comment by ahartikainen** [comment]
This PR can wait to next release. It will change many io functions.

**Review by ahartikainen — APPROVED** [review]

**Pull request merged**